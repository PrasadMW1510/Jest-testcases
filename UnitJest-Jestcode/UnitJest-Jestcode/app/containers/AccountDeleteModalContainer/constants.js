/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
