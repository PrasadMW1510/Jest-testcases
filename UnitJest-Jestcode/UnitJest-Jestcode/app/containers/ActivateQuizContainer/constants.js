/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
