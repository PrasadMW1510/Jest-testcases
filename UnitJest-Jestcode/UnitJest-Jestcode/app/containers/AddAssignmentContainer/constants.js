/*
 *
 * AddAssignmentContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/AddAssignmentContainer/DEFAULT_ACTION';
export const GET_STUDENTDETAIL_SUCCESS = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENTDETAIL_FAILURE = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_STUDENT_DETAILS = 'app/AddAssignmentContainer/GET_STUDENT_DETAILS';
export const POST_NEW_ASSIGNMENT = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT';
export const POST_NEW_ASSIGNMENT_SUCCESS = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS';
