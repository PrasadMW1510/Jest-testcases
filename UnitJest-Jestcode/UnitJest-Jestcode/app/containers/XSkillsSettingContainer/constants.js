/*
 *
 * XSkillsSettingContainer constants
 *
 */

export const XSKILLS_SETTING_CONTAINER_REQUEST =
  'app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST';

export const XSKILLS_SETTING_CONTAINER_REQUEST_FAILURE =
  'app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST_FAILURE';

export const XSKILLS_SETTING_CONTAINER_REQUEST_SUCCESS =
  'app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST_SUCCESS';

export const XSKILLS_SETTINGS_GET_COMMAND = 'get_xskills_settings';

export const XSKILLS_SETTINGS_LOADING = 'app/XSkillsSettingContainer/XSKILLS_SETTINGS_LOADING';

export const XSKILLS_SETTINGS_REQUEST = 'app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST';

export const XSKILLS_SETTINGS_REQUEST_FAILURE =
  'app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST_FAILURE';

export const XSKILLS_SETTINGS_REQUEST_SUCCESS =
  'app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST_SUCCESS';

export const XSKILLS_SETTINGS_SAVE_REQUEST =
  'app/XSkillsSettingContainer/XSKILLS_SETTINGS_SAVE_REQUEST';

export const XSKILLS_SETTINGS_URL = '/xSkills/XTProductCtrls';

export const XSKILLS_TEST_ASSIGN_URL = '/xSkills/XTProductCtrls';

export const XSKILLS_TEST_ASSIGN_GET_COMMAND = 'get_xskills_tests';

export const XSKILLS_TEST_ASSIGN_SET_COMMAND = 'set_xskills_current_test';

export const XSKILLS_TEST_ASSIGNMENT_LOADING = 'XSKILLS_TEST_ASSIGNMENT_LOADING';

export const XSKILLS_TEST_ASSIGNMENT_REQUEST = 'XSKILLS_TEST_ASSIGNMENT_REQUEST';

export const XSKILLS_TEST_ASSIGNMENT_REQUEST_SUCCESS = 'XSKILLS_TEST_ASSIGNMENT_REQUEST_SUCCESS';

export const XSKILLS_TEST_ASSIGNMENT_REQUEST_FAILURE = 'XSKILLS_TEST_ASSIGNMENT_REQUEST_FAILURE';

export const XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST = 'XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST';

export const XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_SUCCESS =
  'XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_SUCCESS';

export const XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_FAILURE =
  'XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_FAILURE';
