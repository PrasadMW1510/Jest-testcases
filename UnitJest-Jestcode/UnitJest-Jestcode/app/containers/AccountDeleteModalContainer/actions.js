/*
 *
 * AccountDeleteModalContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteRequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteMIARequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_MIA_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete is a success
 *
 *  @returns {{type}}
 */
export function postAccountDeleteRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account delete is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountDeleteRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for account unenroll action
 *
 * @param accountUnenrollPayload
 * @returns {{type}}
 */
export function postAccountUnenrollRequest(accountUnenrollPayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST,
    payload: accountUnenrollPayload,
  };
}

/**
 * Handles API logic for account unenroll is a success
 *
 *  @returns {{type}}
 */
export function postAccountUnenrollRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account unenroll is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountUnenrollRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_FAILURE,
    error,
  });
}
