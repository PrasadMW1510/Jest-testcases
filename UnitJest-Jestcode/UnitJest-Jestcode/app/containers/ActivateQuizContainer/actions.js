/*
 *
 * ActivateQuizContainer actions
 *
 */

import * as Constants from './constants';

export function activateQuizRequest(activateQuizPreviewData, searchOpts, searchDataChecked) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST,
    activateQuizPreviewData,
    searchOpts,
    searchDataChecked,
  };
}

export function getPostActivateRequestFailure(error) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST_FAILURE,
    error,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}

export function initialiseModalFlag() {
  return {
    type: Constants.INITIALIZE_MODAL_FLAG,
  };
}
