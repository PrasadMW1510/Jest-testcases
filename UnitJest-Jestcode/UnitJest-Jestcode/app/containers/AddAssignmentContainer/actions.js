/*
 *
 * AddAssignmentContainer actions
 *
 */

import * as Constants from './constants';

export function getStudentDetails(data) {
  return {
    type: Constants.GET_STUDENT_DETAILS,
    data,
  };
}
export function getStudentDetailsSuccess(data) {
  return {
    type: Constants.GET_STUDENTDETAIL_SUCCESS,
    data,
  };
}

export function getStudentDetailsFailure(error) {
  return {
    type: Constants.GET_STUDENTDETAIL_FAILURE,
    error,
  };
}

export function postSaveNewAssignment(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT,
    data,
  };
}

export function postSaveNewAssignmentSuccess(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT_SUCCESS,
    data,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
