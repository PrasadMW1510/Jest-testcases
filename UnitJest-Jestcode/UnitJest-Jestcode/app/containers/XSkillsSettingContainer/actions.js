/*
 *
 * XSkillsSettingContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

export function xSkillsSettingContainerRequest() {
  return {
    type: Constants.XSKILLS_SETTING_CONTAINER_REQUEST,
  };
}

/**
 *
 * Handles API logic for getting XSkills programSettings, testAssignmentMeta are successful
 *
 * @param programSettings, testAssignmentMeta
 * @returns {{type: *, programSetting: {}}}
 */
export function xSkillsSettingContainerRequestSuccess(
  programSettings = {},
  testAssignmentMeta = {}
) {
  return {
    type: Constants.XSKILLS_SETTING_CONTAINER_REQUEST_SUCCESS,
    programSettings,
    testAssignmentMeta,
  };
}

export function xSkillsSettingContainerRequestFailure(error) {
  return generalFailure({
    type: Constants.XSKILLS_SETTING_CONTAINER_REQUEST_FAILURE,
    error,
  });
}

// ############### TEST ASSIGNMENT CODE
export function xSkillsTestAssignmentRequest() {
  return {
    type: Constants.XSKILLS_TEST_ASSIGNMENT_REQUEST,
  };
}

/**
 *
 * Handles API logic for getting XSkills testAssignmentMeta is successful
 *
 * @param testAssignmentMeta
 * @returns {{type: *, testAssignmentMeta: {}}}
 */
export function xSkillsTestAssignmentRequestSuccess(testAssignmentMeta = {}) {
  return {
    type: Constants.XSKILLS_TEST_ASSIGNMENT_REQUEST_SUCCESS,
    testAssignmentMeta,
  };
}

export function xSkillsTestAssignmentRequestFailure(error) {
  return generalFailure({
    type: Constants.XSKILLS_TEST_ASSIGNMENT_REQUEST_FAILURE,
    error,
  });
}

export function xSkillsTestAssignmentLoading() {
  return {
    type: Constants.XSKILLS_TEST_ASSIGNMENT_LOADING,
  };
}

export function xSkillsTestAssignmentSaveRequest(data) {
  return {
    type: Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST,
    data,
  };
}

export function xSkillsTestAssignmentSaveRequestSuccess(results = {}) {
  return {
    type: Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_SUCCESS,
    results,
  };
}

export function xSkillsTestAssignmentSaveRequestFailure(error) {
  return generalFailure({
    type: Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_FAILURE,
    error,
  });
}
// #### END ##### TEST ASSIGNMENT CODE  ##########

// ############### PROGRAM SETTINGS CODE  ########

export function xSkillsSettingsLoading() {
  return {
    type: Constants.XSKILLS_SETTINGS_LOADING,
  };
}

export function xSkillsSettingsRequest() {
  return {
    type: Constants.XSKILLS_SETTINGS_REQUEST,
  };
}

/**
 *
 * Handles API logic for getting XSkills Program Settings is successful
 *
 * @param settings
 * @returns {{type: *, settings: {}}}
 */
export function xSkillsSettingsRequestSuccess(settings = {}) {
  return {
    type: Constants.XSKILLS_SETTINGS_REQUEST_SUCCESS,
    settings,
  };
}

export function xSkillsSettingsRequestFailure(error) {
  return generalFailure({
    type: Constants.XSKILLS_SETTINGS_REQUEST_FAILURE,
    error,
  });
}

export function xSkillsSettingsSaveRequest(settings) {
  return {
    type: Constants.XSKILLS_SETTINGS_SAVE_REQUEST,
    settings,
  };
}

// ##### END ###### PROGRAM SETTINGS CODE  ########
