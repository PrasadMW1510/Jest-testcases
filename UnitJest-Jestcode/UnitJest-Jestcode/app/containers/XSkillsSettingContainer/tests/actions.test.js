import { GENERAL_FAILURE } from 'containers/App/constants';
import * as Actions from '../actions';
import * as Constants from '../constants';

describe('XSkillsSettingContainer actions', () => {
  const mockError = 'mockError';
  const mockSaveData = { output: { output_data: { test_number: '3' } } };
  const mockSaveSettingsData = { settings: [] };
  it('should match xSkillsSettingContainerRequest action', () => {
    const action = Actions.xSkillsSettingContainerRequest();
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_SETTING_CONTAINER_REQUEST);
  });

  it('should match xSkillsSettingContainerRequestSuccess action', () => {
    const action = Actions.xSkillsSettingContainerRequestSuccess();
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_SETTING_CONTAINER_REQUEST_SUCCESS);
  });

  it('should match xSkillsSettingContainerRequestFailure action', () => {
    const action = Actions.xSkillsSettingContainerRequestFailure(mockError);
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(GENERAL_FAILURE);
    expect(action.subType).toEqual(Constants.XSKILLS_SETTING_CONTAINER_REQUEST_FAILURE);
  });

  it('should match xSkillsTestAssignmentSaveRequest action', () => {
    const action = Actions.xSkillsTestAssignmentSaveRequest(mockSaveData);
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST);
  });

  it('should match xSkillsTestAssignmentSaveRequestSuccess action', () => {
    const action = Actions.xSkillsTestAssignmentSaveRequestSuccess();
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_SUCCESS);
  });

  it('should match xSkillsTestAssignmentSaveRequestFailure action', () => {
    const action = Actions.xSkillsTestAssignmentSaveRequestFailure(mockError);
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(GENERAL_FAILURE);
    expect(action.subType).toEqual(Constants.XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_FAILURE);
  });

  it('should match xSkillsSettingsRequest action', () => {
    const action = Actions.xSkillsSettingsRequest();
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_SETTINGS_REQUEST);
  });

  it('should match xSkillsSettingsRequestSuccess action', () => {
    const action = Actions.xSkillsSettingsRequestSuccess({});
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_SETTINGS_REQUEST_SUCCESS);
  });

  it('should match xSkillsSettingsRequestFailure action', () => {
    const action = Actions.xSkillsSettingsRequestFailure(mockError);
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(GENERAL_FAILURE);
    expect(action.subType).toEqual(Constants.XSKILLS_SETTINGS_REQUEST_FAILURE);
  });

  it('should match xSkillsSettingsSaveRequest action', () => {
    const action = Actions.xSkillsSettingsSaveRequest(mockSaveSettingsData);
    expect(action).toMatchSnapshot();
    expect(action.type).toEqual(Constants.XSKILLS_SETTINGS_SAVE_REQUEST);
  });
});
