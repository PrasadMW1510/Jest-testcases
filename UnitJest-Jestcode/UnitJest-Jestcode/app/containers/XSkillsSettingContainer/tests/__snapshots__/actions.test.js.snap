// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XSkillsSettingContainer actions should match xSkillsSettingContainerRequest action 1`] = `
Object {
  "type": "app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingContainerRequestFailure action 1`] = `
Object {
  "error": "mockError",
  "subType": "app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST_FAILURE",
  "type": "general_failure",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingContainerRequestSuccess action 1`] = `
Object {
  "programSettings": Object {},
  "testAssignmentMeta": Object {},
  "type": "app/XSkillsSettingContainer/XSKILLS_SETTING_CONTAINER_REQUEST_SUCCESS",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingsRequest action 1`] = `
Object {
  "type": "app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingsRequestFailure action 1`] = `
Object {
  "error": "mockError",
  "subType": "app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST_FAILURE",
  "type": "general_failure",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingsRequestSuccess action 1`] = `
Object {
  "settings": Object {},
  "type": "app/XSkillsSettingContainer/XSKILLS_SETTINGS_REQUEST_SUCCESS",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsSettingsSaveRequest action 1`] = `
Object {
  "settings": Object {
    "settings": Array [],
  },
  "type": "app/XSkillsSettingContainer/XSKILLS_SETTINGS_SAVE_REQUEST",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsTestAssignmentSaveRequest action 1`] = `
Object {
  "data": Object {
    "output": Object {
      "output_data": Object {
        "test_number": "3",
      },
    },
  },
  "type": "XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsTestAssignmentSaveRequestFailure action 1`] = `
Object {
  "error": "mockError",
  "subType": "XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_FAILURE",
  "type": "general_failure",
}
`;

exports[`XSkillsSettingContainer actions should match xSkillsTestAssignmentSaveRequestSuccess action 1`] = `
Object {
  "results": Object {},
  "type": "XSKILLS_TEST_ASSIGNMENT_SAVE_REQUEST_SUCCESS",
}
`;
